components/slider.js

```diff
@@ -1,23 +1,63 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
-import animateScrollTo from 'animated-scroll-to';
+import scrollToWithAnimation from 'scrollto-with-animation';
 import { Highlight } from '../components/highlights'
 import stylesheet from './slider.scss'
 import { ArrowButton } from './arrow-button';
 import { MediaList } from './icon';
 import { IndustriesMain } from './industries-main';
 
+const autoSliderTime = 15000;
+
 export class Slider extends React.Component {
   constructor(props) {
     super(props);
     this.state = {currentScroll: 0, isBeginning: true, isEnd: false}
     this.scroll = this.scroll.bind(this);
+    this.scrollAutomatically = this.scrollAutomatically.bind(this);
     this.list = this.props.listRef;
+    this.shouldScrollRight = true;
 
     this.setListRef = element => {
       this.list = element;
     };
+
+    if(this.props.shouldScrollAutomatically){
+      this.scrollAutomatically();
+    }
+  }
+
+  scrollAutomatically(){
+    const element = this.list;
+    if(!element){
+      setTimeout(() => this.scrollAutomatically(), 500);
+      return;
+    }
+    const scrollWidth = element.scrollWidth;
+
+    if(this.shouldScrollRight){
+      scrollToWithAnimation(
+        element,
+        'scrollLeft',
+        scrollWidth,
+        autoSliderTime, 
+        'linearTween',      
+      );
+    }
+    else {
+      scrollToWithAnimation(
+        element,
+        'scrollLeft',
+        scrollWidth * -1, 
+        autoSliderTime,
+        'linearTween',
+     );
+    }
+
+    this.shouldScrollRight = !this.shouldScrollRight;
+
+    setTimeout(() => this.scrollAutomatically(), autoSliderTime * 0.65);
```

Comment:
So, my worries with this is that you are setting a recursive loop within your function. I'm not following why this is needed but maybe you can explain the logic. Ideally our code should be descriptive enough so anyone can know what it is doing.

Reply:
Right. So the idea is that media first scrolls in one direction and then the other. So in code terms that means telling it to scroll right, which can take like 10 seconds and then at the end it scrolls left. Etc.